server_environment/models/server_env_tech_name_mixin.py

```diff
@@ -25,41 +26,36 @@ class ServerEnvTechNameMixin(models.AbstractModel):
     _sql_constraints = [
         ("tech_name_uniq", "unique(tech_name)", "`tech_name` must be unique!",)
     ]
-    # TODO: could leverage the new option for computable / writable fields
-    # and get rid of some onchange / read / write code.
+
     tech_name = fields.Char(
+        string="Technical Name",
         help="Unique name for technical purposes. Eg: server env keys.",
+        compute="_compute_tech_name",
+        store=True,
+        readonly=False,
     )
 
     _server_env_section_name_field = "tech_name"
 
-    @api.onchange("name")
-    def _onchange_name_for_tech(self):
-        # Keep this specific name for the method to avoid possible overrides
-        # of existing `_onchange_name` methods
-        if self.name and not self.tech_name:
-            self.tech_name = self.name
-
-    @api.onchange("tech_name")
-    def _onchange_tech_name(self):
-        if self.tech_name:
-            # make sure is normalized
-            self.tech_name = self._normalize_tech_name(self.tech_name)
-
-    @api.model
-    def create(self, vals):
-        self._handle_tech_name(vals)
-        return super().create(vals)
-
-    def write(self, vals):
-        self._handle_tech_name(vals)
-        return super().write(vals)
-
-    def _handle_tech_name(self, vals):
-        # make sure technical names are always there
-        if not vals.get("tech_name") and vals.get("name"):
-            vals["tech_name"] = self._normalize_tech_name(vals["name"])
+    @api.depends("name")
+    def _compute_tech_name(self):
+        for rec in self:
+            # Update tech_name only if it hasn't been set or if we're
+            # dealing with a new record.
+            if not rec.tech_name or not rec._origin.id:
```

Comment:
I'd say only if is not set

Reply:
My reasoning is that on new records (before they're saved) you can type a name: "My paiment providr", and then you realize you typed it incorrectly, so you correct it: "My payment provider".

I would expect the tech_name to update acordingly ("my_payment_provider"), at least until I save it.
After save, yes, it should never change again unless it's explicit.